### PR TITLE
Add Testing code convention

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

This adds Code convention and naming guidelines for test classes, and starts conforming existing tests to the convention.
Not all tests were updated though, and will be progressively through the next few months.

### Motivation

Make the test classes easier to read and maintain.

### CODE REVIEW

File: dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/net/info/BroadcastReceiverNetworkInfoProviderTest.kt [hunk 1]
```diff
@@ -84,20 +85,20 @@ internal class BroadcastReceiverNetworkInfoProviderTest {
 
     @Test
     fun `it will do nothing if unregister is called before register`() {
-        // when
+        // When
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/net/info/BroadcastReceiverNetworkInfoProviderTest.kt:88]: You forgot to change the methods name but I guess we could do this in another PR.

Comment (xgouchet) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/net/info/BroadcastReceiverNetworkInfoProviderTest.kt:88]: This was a simple Search and replace, I didn't do it manually :D
